### DISCUSSION TITLE
Where to put custom plugins when install smarty with composer

### DISCUSSION THREAD

**Discussion by iamaks1993** [smarty-php/smarty#948] [Q&A] [answered]
Hi, I hope you are doing well.

Previously, we were on smarty version 4.0.0 In that, we have created a few plugins. Now, we are planning to migrate from the current version to 4.4.1 by using composer. So, my question is where to store custom plugins when using composer. Thanks in advance.

**Comment by wisskid** [comment, marked as answer]
Put them anywhere you like and then use [addPluginsDir](https://smarty-php.github.io/smarty/4.x/programmers/api-functions/api-add-plugins-dir/) to load the entire directory!

Or register individual plugins using registerPlugin.
